Sources/StreamChatUI/ImageGallery/ImageGalleryVC.swift

```diff
@@ -0,0 +1,305 @@
+        
+        view.embed(attachmentsCollectionView)
+        
+        view.addSubview(topBarView)
+        topBarView.pin(anchors: [.leading, .trailing, .top], to: view)
+        
+        let topBarContainerStackView = ContainerStackView()
+            .withoutAutoresizingMaskConstraints
+        topBarView.embed(topBarContainerStackView)
+        topBarContainerStackView.preservesSuperviewLayoutMargins = true
+        topBarContainerStackView.isLayoutMarginsRelativeArrangement = true
+        
+        topBarContainerStackView.addArrangedSubview(closeButton)
+        
+        let infoContainerStackView = ContainerStackView()
+        infoContainerStackView.axis = .vertical
+        infoContainerStackView.alignment = .center
+        infoContainerStackView.spacing = 4
+        topBarContainerStackView.addArrangedSubview(infoContainerStackView)
+        infoContainerStackView.pin(anchors: [.centerX], to: view)
+        
+        userLabel.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        infoContainerStackView.addArrangedSubview(userLabel)
+        
+        infoContainerStackView.addArrangedSubview(dateLabel)
+        
+        topBarContainerStackView.addArrangedSubview(UIView.spacer(axis: .horizontal))
+        
+        view.addSubview(bottomBarView)
+        bottomBarView.pin(anchors: [.leading, .trailing, .bottom], to: view)
+        
+        let bottomBarContainerStackView = ContainerStackView()
+            .withoutAutoresizingMaskConstraints
+        bottomBarContainerStackView.preservesSuperviewLayoutMargins = true
+        bottomBarContainerStackView.isLayoutMarginsRelativeArrangement = true
+        bottomBarView.embed(bottomBarContainerStackView)
+        
+        shareButton.setContentHuggingPriority(.streamRequire, for: .horizontal)
+        bottomBarContainerStackView.addArrangedSubview(shareButton)
+
+        currentPhotoLabel.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        bottomBarContainerStackView.addArrangedSubview(currentPhotoLabel)
+        currentPhotoLabel.pin(anchors: [.centerX], to: view)
+        
+        bottomBarContainerStackView.addArrangedSubview(.spacer(axis: .horizontal))
+    }
+    
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        
+        attachmentsCollectionView.layoutIfNeeded()
+
+        let initialPage = images
+            .firstIndex(where: { $0.id == initialAttachment.id }) ?? 0
+        currentPage = initialPage
+        let contentOffset = CGPoint(
+            x: attachmentsCollectionView.bounds.width * CGFloat(currentPage),
+            y: attachmentsCollectionView.contentOffset.y
+        )
+        attachmentsCollectionView.setContentOffset(contentOffset, animated: false)
+    }
+    
+    override open func updateContent() {
+        super.updateContent()
+        
+        images = content.attachments
+            .filter { $0.type == .image }
+            .compactMap { $0 as? ChatMessageDefaultAttachment }
```

Comment:
Since `updateContent` can be called multiple times, should we reload the CW when assigning the new `images` value?